bundles/org.openhab.binding.kaleidescape/src/main/java/org/openhab/binding/kaleidescape/internal/handler/KaleidescapeHandler.java

```diff
@@ -0,0 +1,617 @@
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String channel = channelUID.getId();
+
+        if (getThing().getStatus() != ThingStatus.ONLINE) {
+            logger.debug("Thing is not ONLINE; command {} from channel {} is ignored", command, channel);
+            return;
+        }
+        synchronized (sequenceLock) {
+            if (!connector.isConnected()) {
+                logger.debug("Command {} from channel {} is ignored: connection not established", command, channel);
+                return;
+            }
+
+            try {
+                if (command instanceof RefreshType) {
+                    handleRefresh(channel);
+                    return;
+                }
+
+                switch (channel) {
+                    case POWER:
+                        if (command instanceof OnOffType) {
+                            connector.sendCommand(command == OnOffType.ON ? LEAVE_STANDBY : ENTER_STANDBY);
+                        }
+                        break;
+                    case VOLUME:
+                        if (command instanceof PercentType) {
+                            this.volume = (int) ((PercentType) command).doubleValue();
+                            logger.debug("Got volume command {}", this.volume);
+                            connector.sendCommand(SEND_EVENT_VOLUME_LEVEL_EQ + this.volume);
+                        }
+                        break;
+                    case MUTE:
+                        if (command instanceof OnOffType) {
+                            this.isMuted = command == OnOffType.ON ? true : false;
+                        }
+                        connector.sendCommand(SEND_EVENT_MUTE + (this.isMuted ? MUTE_ON : MUTE_OFF));
+                        break;
+                    case MUSIC_REPEAT:
+                        if (command instanceof OnOffType) {
+                            connector.sendCommand(command == OnOffType.ON ? MUSIC_REPEAT_ON : MUSIC_REPEAT_OFF);
+                        }
+                        break;
+                    case MUSIC_RANDOM:
+                        if (command instanceof OnOffType) {
+                            connector.sendCommand(command == OnOffType.ON ? MUSIC_RANDOM_ON : MUSIC_RANDOM_OFF);
+                        }
+                        break;
+                    case CONTROL:
+                    case MUSIC_CONTROL:
+                        handleControlCommand(command);
+                        break;
+                    default:
+                        logger.debug("Command {} from channel {} failed: unexpected command", command, channel);
+                        break;
+                }
+            } catch (KaleidescapeException e) {
+                logger.warn("Command {} from channel {} failed: {}", command, channel, e.getMessage());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Sending command failed");
```

Comment:
If you also set the status this can be debug. You could set the command and message in the status to give the user this information directly.
```suggestion
                logger.debug("Command {} from channel {} failed: {}", command, channel, e.getMessage());
                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Sending command failed");
```